lib/processors/manifestCreator.js

```diff
@@ -525,6 +560,61 @@ async function createManifest(libraryResource, libBundle, descriptorVersion, _in
 		return sapUI5;
 	}
 
+	/**
+	 * Creates an i18n section:
+	 * - either using bundleUrl and supportedLocales
+	 * - or the i18n String
+	 *
+	 * @param {string} i18n bundle url, e.g. "messagebundle.properties"
+	 * @param {Map<string, Set<string>>} i18nToSupportedLocales cache to determine the supportedLocales only once
+	 * @returns {{bundleUrl: string, supportedLocales: string[]}|null|string} json structure with bundleUrl and
+	 *   supportedLocales or the i18n String if not a ".properties" file.
+	 *   <code>null</code> if given i18n String is <code>null</code>
+	 */
+	function createI18nSection(i18n, i18nToSupportedLocales) {
+		if (!i18n) {
+			return null;
+		}
+		if (!i18n.endsWith(".properties")) {
+			return i18n;
+		}
+
+		// if the supported locales information should not be included use i18n text
+		if (!includeSupportedLocalesInformation) {
+			return i18n;
+		}
+
+		let supportedLocales = i18nToSupportedLocales.get(i18n);
+
+		if (!supportedLocales) {
+			supportedLocales = new Set();
+
+			if (libBundle.findResource(i18n) != null) {
+				supportedLocales.add("");
+			}
+			const i18nPatternStr = i18n.substring(0, i18n.length - ".properties".length);
+			const i18nRegexp = new RegExp(i18nPatternStr + "_([^.]+)\\.properties$");
```

Comment:
Does `i18nPatternStr` need to be encoded for RegExp or can we be sure that it doesn't contain special characters?

Reply:
tackled with: f649422